### ISSUE TITLE
Error squiggles don't show up in files with the C extension

### ISSUE THREAD

**Issue by jonstvns** [microsoft/vscode-cpptools#6000]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10 Pro Version 10.0.18363 Build 18363
- VS Code Version: 1.48.0
- C/C++ Extension Version: 0.29.0
- Other extensions you installed (and if the issue persists after disabling them): None
- Does this issue involve using SSH remote to run the extension on a remote machine?: No
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

Error squiggles do not show up in the editor in C files, but renaming the files to be CPP files restores the error functionality.

This is a single file project building for msvc x64.

**Steps to reproduce**

1. Open a file with the "c" file extension
2. Write an intentional syntax error
3. No error squiggle shows up

4. Rename file with the "cpp" file extension
5. Error squiggles show up

**Expected behavior**

Error squiggles show up in the editor for both c and cpp files

<details>
  <summary><strong> Logs </strong></summary>

```
-------- Diagnostics - 8/21/2020, 5:42:26 AM
Version: 0.29.0
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/src/**",
        "${workspaceFolder}/external/**"
    ],
    "browse": {
        "limitSymbolsToIncludedHeaders": true,
        "path": [
            "${workspaceFolder}/src/**",
            "${workspaceFolder}/external/**"
        ]
    },
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE",
        "GRAPHICS_API_OPENGL_33",
        "PLATFORM_DESKTOP"
    ],
    "windowsSdkVersion": "10.0.18362.0",
    "compilerPath": "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe",
    "cStandard": "c11",
    "cppStandard": "c++17",
    "intelliSenseMode": "msvc-x64",
    "compilerArgs": []
}
Translation Unit Mappings:
[ C:\Work\LanguageGame\game\src\main.cpp ]:
    C:\WORK\LANGUAGEGAME\GAME\SRC\MAIN.CPP
Translation Unit Configurations:
[ C:\Work\LanguageGame\game\src\main.cpp ]:
    Process ID: 12940
    Memory Usage: 13 MB
    Compiler Path: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.27.29110/bin/Hostx64/x64/cl.exe
    Includes:
        C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\COMMUNITY\VC\TOOLS\MSVC\14.27.29110\INCLUDE
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\UM
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\UCRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\SHARED
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\WINRT
        C:\PROGRAM FILES (X86)\WINDOWS KITS\10\INCLUDE\10.0.18362.0\CPPWINRT
        C:\WORK\LANGUAGEGAME\GAME\EXTERNAL\RAYLIB\INCLUDE
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
        GRAPHICS_API_OPENGL_33
        PLATFORM_DESKTOP
    Standard Version: ms_c++17
    IntelliSense Mode: msvc-x64
Total Memory Usage: 13 MB


```
</details>

**Screenshots**

C file without error squiggles
![image](https://user-images.githubusercontent.com/1030863/90892655-7e1f9f80-e372-11ea-88e4-4da0c8b4a26d.png)

CPP file with error squiggles
![image](https://user-images.githubusercontent.com/1030863/90892732-a3141280-e372-11ea-9791-db82fe6dd099.png)

**Comment by sean-mcmanus** [comment]
"undeclared identifier" is not an error from our extension. Where is that coming from? I'm not able to repro the bug, so it might be caused by another extension you have installed.

**Comment by jonstvns** [comment]
I discovered this only happens if you open the folder or workspace directly instead of through a Developer Command Prompt.

I added this to my code-workspace settings so I can open the workspace without the need to open a separate commandline first. 
````
"settings": {
	"terminal.integrated.shell.windows": "C:\\Windows\\system32\\cmd.exe",
	"terminal.integrated.shellArgs.windows": [
		"C:\\Program^ Files^ ^(x86^)\\Microsoft^ Visual^ Studio\\2019\\Community\\Common7\\Tools\\VsDevCmd.bat"
	]
}
````

Errors now show up in main.c as expected when I open my workspace.

For the record, no other extensions are installed. This was a fresh vscode installation.
